TTL-Modifier, average ttl-selector crashes

When using `ttl-select = "average"` on `ttl-modifier`, it generates a "divide by zero" crash.

Looks like when the answer section is empty, because of an empty `noerror` response for example. it crashes. Seems also to happen on any responses other then `noerror` (like `nxdomain`) as well.
```
panic: runtime error: integer divide by zero

goroutine 65 [running]:
github.com/folbricht/routedns.TTLSelectAverage(0x14000225c20?, 0x1400039e870?)
        /opt/routedns/build/routedns/ttl-modifier.go:131 +0xac
github.com/folbricht/routedns.(*TTLModifier).Resolve(0x140002364b0, 0x1400039e870?, {{0x14000400000, 0x4, 0x4}, {0x0, 0x0}, {0x0, 0x0}, {0x140001d0028, ...}})
        /opt/routedns/build/routedns/ttl-modifier.go:58 +0x9c
github.com/folbricht/routedns.(*Cache).Resolve(0x14000233500, 0x1400039e750, {{0x14000400000, 0x4, 0x4}, {0x0, 0x0}, {0x0, 0x0}, {0x140001d0028, ...}})
        /opt/routedns/build/routedns/cache.go:121 +0x3e4
github.com/folbricht/routedns.(*EDNS0Modifier).Resolve(0x140002365d0?, 0x1400039e750?, {{0x14000400000, 0x4, 0x4}, {0x0, 0x0}, {0x0, 0x0}, {0x140001d0028, ...}})
        /opt/routedns/build/routedns/edns0-modifier.go:39 +0xc4
github.com/folbricht/routedns.(*requestDedup).Resolve(0x14000236690, 0x1400039e750, {{0x14000400000, 0x4, 0x4}, {0x0, 0x0}, {0x0, 0x0}, {0x140001d0028, ...}})
        /opt/routedns/build/routedns/request-dedup.go:108 +0x430
github.com/folbricht/routedns.(*ClientBlocklist).Resolve(0x14000233570, 0x1400039e750, {{0x14000400000, 0x4, 0x4}, {0x0, 0x0}, {0x0, 0x0}, {0x140001d0028, ...}})
        /opt/routedns/build/routedns/client-blocklist.go:57 +0x4f4
github.com/folbricht/routedns.listenHandler.func1({0x105131aa8, 0x14000380180}, 0x1400039e750)
        /opt/routedns/build/routedns/dnslistener.go:80 +0x5ec
github.com/miekg/dns.HandlerFunc.ServeDNS(0x140001b6400?, {0x105131aa8?, 0x14000380180?}, 0x140001b6400?)
        /opt/routedns/build/golang/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:37 +0x38
github.com/miekg/dns.(*Server).serveDNS(0x14000242240, {0x140001b6400, 0x25, 0x200}, 0x14000380180)
        /opt/routedns/build/golang/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:659 +0x3a0
github.com/miekg/dns.(*Server).serveUDPPacket(0x14000242240, 0x0?, {0x140001b6400, 0x25, 0x200}, {0x10512fa30?, 0x140000a8978}, 0x14000404000, {0x0?, 0x0})
        /opt/routedns/build/golang/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:603 +0x1b4
created by github.com/miekg/dns.(*Server).serveUDP
        /opt/routedns/build/golang/pkg/mod/github.com/miekg/dns@v1.1.50/server.go:533 +0x3f0
```

--- a/ttl-modifier.go
+++ b/ttl-modifier.go
@@ -128,6 +128,10 @@ func TTLSelectAverage(r *TTLModifier, a *dns.Msg) bool {
 		n++
 		sum += int(h.Ttl)
 	})
+	if n == 0 {
+		// Avoid division by 0 for empty responses
+		n = 1
+	}
 	average := uint32(sum / n)
 	iterateOverAnswerRRHeader(a, func(h *dns.RR_Header) {
 		if h.Ttl != average {

